build-system/common/transform-cache.js

```diff
@@ -104,7 +104,7 @@ const readCache = new Map();
  *
  * @param {string} path
  * @param {string=} optionsHash
- * @return {{contents: string, hash: string}}
+ * @return {Promise<{contents: string, hash: string}>}
```

Comment:
This type had been incorrect

Reply:
Good catch.

@rileyajones Any idea why type-checking couldn't catch this? Could it be that the `Promise<>` part is inferred by Typescript?

Reply:
Separate comment: Line 96 uses the type `{hash: string, contents: string}`, while this line uses a type with the keys in the opposite order. Would it make sense to declare a typedef and use it everywhere?

Reply:
complete guess: maybe the js-based type checking is less intelligent and gives more trust to the stated types?

Reply:
@rsimha It looks like typescript is not picking up the type declaration for `readCache` and thus assumes its type is `Map<any, any>`.

I would suggest explicitly declaring the type of `readCache`
```
/**
 * Used to cache file reads, since some (esbuild) will have multiple "loads" per
 * file. This batches consecutive reads into a single, and then clears its cache
 * item for the next load.
 * @private @const {!Map<string, Promise<{hash: string, contents: string}>>}
 * @type {!Map<string, Promise<{hash: string, contents: string}>>}
 */
const readCache = new Map();
```

Reply:
Done